packages/web-components/src/components/rux-timeline/rux-track/rux-track.tsx

```diff
@@ -154,16 +154,20 @@ export class RuxTrack {
                 el.timezone = this.timezone
                 el.style.gridRow = '1'
                 el.style.visibility = 'inherit'
-                const gridColumn = `${this.calculateGridColumnFromTime(
+                const gridColumn = `${this._calculateGridColumnFromTime(
                     el.start
-                )} / ${this.calculateGridColumnFromTime(el.end)}`
+                )} / ${this._calculateGridColumnFromTime(el.end)}`
                 el.style.gridColumn = gridColumn
             } else {
                 if (!isHidden) {
                     el.style.visibility = 'hidden'
                     console.error(isValid.error)
                 }
             }
+
+            el.addEventListener('ruxtimeregionchange', () => {
+                this.initializeRows()
```

Comment:
Will this reattach the handler every time there is a region change? If so, it might start slowing down as the component is used more.

FWIW, I do like the choice of events in this. My understanding is that browsers have made non-bubbling events very performant, plus they are still synchronous, and for added niceness the listeners can throw without stopping other handlers or the functions that dispatch them.

Reply:
Good catch. They were in fact reattaching. 

My quick solution: attach the listener on the track host instead of each region element using Stencils @Listen sugar. 

alternative: I could set the ruxtimeregionchange to not bubble and go the same approach I had originally, but instead either 
1. Remove the event listener and add it every time (probably not performant?)
2. Manage the state of whether a listener has already been attached internally.